### ISSUE TITLE
Add better return types for toJson methods

### ISSUE THREAD

**Issue by stephenh** [stephenh/ts-proto#494]
See #492 but it'd be nice to `.toJson()` to return `JsonOf<FooMessage>` or `FooMessageJson` so that API layers that want "basically JSON, but with a few custom mappings" could reuse `.toJson()` instead of missing out b/c of the `unknown`.

**Comment by boukeversteegh** [comment]
Just adding some thoughts: Would `JsonOf` be new generic type? In several cases, the generated Json has a different interface than the native typescript interface, so I guess this implies that a generic type would not work for all cases. It may require a dedicated interface per message, or only for those cases where there are canonical representations.

**Comment by stephenh** [comment]
@boukeversteegh yeah, good point that, if `JsonOf` was a mapped type, it would need to have conditionals to recognize the various "this TS native type --> this one-off JSON type"...

Which, yeah, that sort of complexity is typically handled better by just doing codegen.
